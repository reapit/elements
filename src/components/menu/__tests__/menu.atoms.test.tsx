import React from 'react'
import { render } from '@testing-library/react'
import { MenuContainer, MenuItem, MenuItemGroup, MenuItemGroupTitle, MenuRadioItem } from '..'

describe('MenuContainer component', () => {
  it('should render MenuContainer and match snapshot', () => {
    const { asFragment } = render(
      <MenuContainer>
        <div>Menu Content</div>
      </MenuContainer>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('MenuItemGroupTitle component', () => {
  it('should render MenuItemGroupTitle and match snapshot', () => {
    const { asFragment } = render(<MenuItemGroupTitle>Group Title</MenuItemGroupTitle>)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('MenuItemGroup component', () => {
  it('should render MenuItemGroup with title and match snapshot', () => {
    const { asFragment } = render(
      <MenuItemGroup title="Group Title">
        <div>Group Item 1</div>
      </MenuItemGroup>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render MenuItemGroup without title and match snapshot', () => {
    const { asFragment } = render(
      <MenuItemGroup>
        <div>Group Item 1</div>
      </MenuItemGroup>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('MenuItem component', () => {
  it('should render Menu Item and match snapshot', () => {
    const { asFragment } = render(<MenuItem>Menu Item</MenuItem>)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('MenuRadioItem component', () => {
  it('should render checked MenuRadioItem and match snapshot', () => {
    const { asFragment } = render(<MenuRadioItem checked>Radio label</MenuRadioItem>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render unchecked MenuRadioItem and match snapshot', () => {
    const { asFragment } = render(<MenuRadioItem>Radio label</MenuRadioItem>)
    expect(asFragment()).toMatchSnapshot()
  })
})
