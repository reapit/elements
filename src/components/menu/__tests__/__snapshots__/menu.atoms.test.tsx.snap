// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuContainer component should render MenuContainer and match snapshot 1`] = `
<DocumentFragment>
  <mock-styled.div>
    <div>
      Menu Content
    </div>
  </mock-styled.div>
</DocumentFragment>
`;

exports[`MenuItem component should render Menu Item and match snapshot 1`] = `
<DocumentFragment>
  <mock-styled.li>
    Menu Item
  </mock-styled.li>
</DocumentFragment>
`;

exports[`MenuItemGroup component should render MenuItemGroup with title and match snapshot 1`] = `
<DocumentFragment>
  <mock-styled.ul>
    <li>
      <mock-styled.div>
        Group Title
      </mock-styled.div>
    </li>
    <div>
      Group Item 1
    </div>
  </mock-styled.ul>
</DocumentFragment>
`;

exports[`MenuItemGroup component should render MenuItemGroup without title and match snapshot 1`] = `
<DocumentFragment>
  <mock-styled.ul>
    <div>
      Group Item 1
    </div>
  </mock-styled.ul>
</DocumentFragment>
`;

exports[`MenuItemGroupTitle component should render MenuItemGroupTitle and match snapshot 1`] = `
<DocumentFragment>
  <li>
    <mock-styled.div>
      Group Title
    </mock-styled.div>
  </li>
</DocumentFragment>
`;

exports[`MenuRadioItem component should render checked MenuRadioItem and match snapshot 1`] = `
<DocumentFragment>
  <mock-styled.li
    aria-selected="true"
    data-selected="true"
    role="option"
  >
    <mock-styled.span
      aria-hidden="true"
      classname="mock-css."
    >
      <svg
        role="img"
        title="Icon image with name check"
      />
    </mock-styled.span>
    Radio label
  </mock-styled.li>
</DocumentFragment>
`;

exports[`MenuRadioItem component should render unchecked MenuRadioItem and match snapshot 1`] = `
<DocumentFragment>
  <mock-styled.li
    role="option"
  >
    Radio label
  </mock-styled.li>
</DocumentFragment>
`;
